lib/spack/spack/cmd/buildcache.py

```diff
@@ -81,6 +81,9 @@ def setup_parser(subparser: argparse.ArgumentParser):
     push_sign.add_argument(
         "--key", "-k", metavar="key", type=str, default=None, help="key for signing"
     )
+    push_sign.add_argument(
+        "--only-key", "-K", metavar="key_to_push", type=str, default=None, help="key for signing"
```

Comment:
I would drop `-K` cause in other places upper case means `--no-*`.

Reply:
I did not realize that, I was torn on including the short option anyway since the `only` part of this is something that you don't want to confuse with  the lowercase `-k` option.